runtime/shared_common/CompositeCache.cpp

```diff
@@ -6886,3 +6913,95 @@ SH_CompositeCacheImpl::isNewCache(void)
 	}
 	return _initializingNewCache;
 }
+
+/**
+ * Record the minimum and maximum boundaries of accessed address in this shared classes cache.
+ * @param [in] metadataAddress address accessed in metadata
+ * @return true if the boundaries are updated, false otherwise
+ */
+bool
+SH_CompositeCacheImpl::updateAccessedShrCacheMetadataBounds(J9VMThread* currentThread, uintptr_t const * metadataAddress)
+{
+	if (!_started) {
+		Trc_SHR_Assert_ShouldNeverHappen();
+		return false;
+	}
+
+	if (!isAddressInMetaDataArea((const void*)metadataAddress)) {
+		return false;
+	}
+
+	uintptr_t* updateAddress = (uintptr_t *) &_minimumAccessedShrCacheMetadata;
+	uintptr_t currentValue = (uintptr_t) _minimumAccessedShrCacheMetadata;
+	uintptr_t const newValue = (uintptr_t const) metadataAddress;
+
+	if (0 == currentValue) { /* set initial value.  Don't care if someone beats us to this  */
+		Trc_SHR_CM_updateAccessedShrCacheMetadataMinimum(currentThread, metadataAddress);
+		compareAndSwapUDATA(
+				updateAddress,
+				currentValue,
+				newValue
+		);
+	}
+
+	currentValue = (uintptr_t) _minimumAccessedShrCacheMetadata;
+	while (newValue < (uintptr_t) currentValue) {
+		Trc_SHR_CM_updateAccessedShrCacheMetadataMinimum(currentThread, metadataAddress);
+
+		compareAndSwapUDATA(
+				updateAddress,
+				currentValue,
+				newValue
+		);
+		currentValue = (uintptr_t) _minimumAccessedShrCacheMetadata;
+	}
+
+	updateAddress = (uintptr_t *) &_maximumAccessedShrCacheMetadata;
+
+	if (0 == currentValue) { /* set initial value.  Don't care if someone beats us to this  */
```

Comment:
This isn't necessary because the `newValue > _maximumAccessedShrCacheMetadata` will always catch this case. Plus it's weird because I don't think currentValue will ever be zero at this point.

Reply:
This line is removed.